pageserver/src/consumption_metrics.rs

```diff
@@ -65,15 +67,24 @@ pub async fn collect_metrics(
         "synthetic size calculation",
         false,
         async move {
+            let started_at = Instant::now();
             calculate_synthetic_size_worker(synthetic_size_calculation_interval, &worker_ctx)
                 .instrument(info_span!("synthetic_size_worker"))
                 .await?;
+            crate::tenant::tasks::warn_when_period_overrun(
+                started_at.elapsed(),
+                synthetic_size_calculation_interval,
+                "synthetic_size_worker",
+            );
             Ok(())
         },
```

Comment:
This one is one level too high. It needs to go inside the loop of `calculate_synthetic_size_worker`.